src/window.ts

```diff
@@ -112,24 +114,66 @@ export type SMonsterEventNotLive = {
 
 export type SMonsterEvent = BetterUXWrapper<SMonsterEventNotLive | SMonsterEventLive>;
 
+export type SEventBooleanKeys =
+  | "egghunt"
+  | "halloween"
+  | "holidayseason"
+  | "lunarnewyear"
+  | "valentines";
+
+export type SPartialEventBoolean = Partial<Record<Extract<SEventBooleanKeys, EventKey>, boolean>>;
+
+// TODO: should we just use MonsterKey?
+export type SPartialEventMonsterEvent = Partial<
+  Record<Exclude<EventKey, SEventBooleanKeys | "abtesting" | "goobrawl">, SMonsterEvent>
+>;
+export type SPartialHolidaySeasonMonsters = Partial<Record<"grinch" | "snowman", SMonsterEvent>>;
+export type SPartialHalloweenMonsters = Partial<
+  Record<"mrpumpkin" | "mrgreen" | "slenderman", SMonsterEvent>
```

Comment:
I think it would make more sense to group all monsters in a single interface, no need to separate them per event, I think.

Maybe not using `Record` would also be more readable like

```ts
interface SEventMonsters {
  // Holiday season
  grinch: SMonsterEvent;
  snowman: SMonsterEvent;

  // Halloween
  mrpumpkin: SMonsterEvent;
  mrgreen: SMonsterEvent;
  slenderman: SMonsterEvent;
}
```

Reply:
The whole point of using record is to allow me to do `parent.S[key]` though 🤔 you can't do that unless there is a record / indexer. Or that was the experience I had at least.

Reply:
That depends on what you want to index with, it has to be a strict subtype of what's available in the object

Reply:
One way I used to use it was like this, for a general purpose join on events, but I guess that could lead to unintended behaviour, considering you can not use `join` on all events 🤔 
![image](https://user-images.githubusercontent.com/9084377/209368554-2162227b-d13c-4b50-94fc-f2e1cf1b17a5.png)

Reply:
Yeah, I see what you mean but not all events are joinable, indeed